VerloopSDK/VerloopSDK.swift

```diff
@@ -137,89 +211,89 @@ import Foundation
+                textColor = VerloopSDK.hexStringToUIColor(hex: clientInfo.textColor)
+                title = clientInfo.title
+                verloopController?.dismissLoader()
+                refreshClientInfo()
+                manager.updateReadyState(true)
+            } catch {
+            }
+            
+            do {
+                   let buttonInfo =  try JSONDecoder().decode(OnButtonClick.self, from: data)
+                   let title = buttonInfo.title ?? ""
+                   let type = buttonInfo.type ?? ""
+                   let payload = buttonInfo.payload ?? ""
+                if type.lowercased() == MessageType.MessageButtonClick.rawValue.lowercased() {
+                    config.getButtonClickListener()?(title,type, payload)
+                } else if type.lowercased() == MessageType.MessageURLClick.rawValue.lowercased() {
+                    config.getURLClickListener()?(buttonInfo.payload)
+                }
+                    
+                    
+                  print("buttton click ")
+                }catch {
+                   print("Problem retreiving button Info \(error)")
+                }
+            
+        }
         
-        window.isOpaque = true
-        window.backgroundColor = UIColor.white
-        window.frame = UIScreen.main.bounds//UIApplication.shared.keyWindow!.frame
-        window.windowLevel = UIWindow.Level.normal + 1
-        window.rootViewController = getNavController()
-        window.makeKeyAndVisible()
-    }
     
+        }
     
+
+
     @objc public func hide() {
-        onChatClose()
-    }
-    
-    func onChatClose() {
-        if previousWindow != nil {
-            window.resignKey()
-            previousWindow!.makeKeyAndVisible()
-            previousWindow = nil
-            
-            window.windowLevel = UIWindow.Level.normal - 30
+        onChatClose {
+            //nothing to do here
         }
     }
     
-    func onTransition() {
-        NSLog("onTransiotion")
-        
+    func onChatClose(completion:@escaping(() -> Void)) {
+        if self.verloopController != nil {
+            self.verloopNavigationController?.dismiss(animated: true, completion: {
+                completion()
+            })
+        } else {
+            completion()
+        }
     }
     
+    //below are the 3 objects which will be prepare as part of the JSCallback delegate methods for button click, urlclick and client info
     private struct ClientInfo: Decodable {
         public let title: String
         public let bgColor: String
         public let textColor: String
     }
     
     private struct OnButtonClick: Decodable {
-        public let title: String
-        public let type: String
-        public let payload: String
+        public let title: String?
+        public let type: String?
+        public let payload: String?
     }
     
     private struct OnURLClick: Decodable {
-        public let url: String
+        public var title:String?
```

Comment:
on url click, we support just the url as string

Reply:
From the looks of it, minor refactoring to be done here. I believe the payload contains the url which is being passed to the callback on the front. Will debug this case subsequently.